File: recommend/models/quotation.py
from sqlalchemy import Column, Integer, String, Date
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class Quotation(Base):
    __tablename__ = 'quotation'
    id = Column(Integer, primary_key=True)
    user_id = Column(UUID(as_uuid=True), comment='견적함')
    cpu_id = Column(Integer)
    power_id = Column(Integer)
    mainboard_id = Column(Integer)
    ram_id = Column(Integer)
    graphic_id = Column(Integer)
    hdd_id = Column(Integer)
    ssd_id = Column(Integer)
    case_id = Column(Integer)
    cooler_id = Column(Integer)
    name = Column(String(100), comment='견적함')
    state = Column(Integer, comment='비트 마스킹, 견적함')
    create_date = Column(Date, comment='견적함')

File: backend/src/main/java/com/windows33/cavendish/domain/quotation/entity/Quotation.java
package com.windows33.cavendish.domain.quotation.entity;

import com.windows33.cavendish.global.converter.DateToStringConverter;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;

@Getter
@SuperBuilder
@NoArgsConstructor
@Table(name = "quotation")
@Entity
public class Quotation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private Integer userId;

    private Integer cpuId;

    private Integer gpuId;

    private Integer ramId;

    private Integer hddId;

    private Integer ssdId;

    private Integer powerId;

    private Integer mainboardId;

    private Integer coolerId;

    private Integer caseId;

    private String name;

    private Integer state;

    @Convert(converter = DateToStringConverter.class)
    @Column(insertable = false)
    private String createDateTime;

    public void removeQuotation() {
        this.state = 1;
    }

    public void updateQuotation(
            Integer cpuId,
            Integer gpuId,
            Integer ramId,
            Integer hddId,
            Integer ssdId,
            Integer powerId,
            Integer mainboardId,
            Integer coolerId,
            Integer caseId,
            String name,
            Integer state
    ) {
        this.cpuId = cpuId;
        this.gpuId = gpuId;
        this.ramId = ramId;
        this.hddId = hddId;
        this.ssdId = ssdId;
        this.powerId = powerId;
        this.mainboardId = mainboardId;
        this.coolerId = coolerId;
        this.caseId = caseId;
        this.name = name;
        this.state = state;
    }

}
